src/httpserver.cpp

```diff
@@ -619,6 +667,12 @@ CService HTTPRequest::GetPeer()
         const char* address = "";
         uint16_t port = 0;
         evhttp_connection_get_peer(con, (char**)&address, &port);
+        if (!strcmp(address, "localhost")) {
+            /* Special: will get here for UNIX sockets. As we have no way to indicate that,
+             * just pass localhost IPv6.
+             */
+            address = "::1";
```

Comment:
+1. This seems like another good use-case for NET_INTERNAL from #10446 (or something similar). As a bonus, that would allow us to use associative containers for unix socket addresses, since each path would have its own hash.